qemu/tests/snp_basic_config.py

```diff
@@ -0,0 +1,75 @@
+import os
+
+from avocado.utils import process
+from virttest import data_dir as virttest_data_dir
+from virttest import env_process, error_context
+from virttest.utils_misc import verify_dmesg
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu snp basic test on Milan and above host:
+    1. Check host snp capability
+    2. Boot snp VM
+    3. Verify snp enabled in guest
+    4. Check snp qmp cmd and policy
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    error_context.context("Start sev-snp test", test.log.info)
+    timeout = params.get_numeric("login_timeout", 240)
+
+    snp_module_path = params["snp_module_path"]
+    if os.path.exists(snp_module_path):
+        f = open(snp_module_path, "r")
+        output = f.read().strip()
+        f.close()
+        if output not in params.objects("module_status"):
+            test.cancel("Host sev-snp support check fail.")
+    else:
+        test.cancel("Host sev-snp support check fail.")
+    if params.get("singlesocket_host_check") == "yes":
+        s, o = process.getstatusoutput(params.get("singlesocket_check_cmd"), shell=True)
+        if s != 1:
+            test.cancel("Host cpu has more than 1 socket, skip the case.")
+
+    vm_name = params["main_vm"]
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    vm.create()
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    verify_dmesg()
+    vm_policy = vm.params.get_numeric("snp_policy")
+    guest_check_cmd = params["snp_guest_check"]
+    sev_guest_info = vm.monitor.query_sev()
+    if sev_guest_info["snp-policy"] != vm_policy:
+        test.fail("QMP snp policy doesn't match %s." % vm_policy)
+    try:
+        session.cmd_output(guest_check_cmd, timeout=240)
+    except Exception as e:
+        test.fail("Guest snp verify fail: %s" % str(e))
+    else:
+        # Verify attestation
+        error_context.context("Start to do attestation", test.log.info)
+        guest_dir = params["guest_dir"]
+        host_script = params["host_script"]
+        guest_cmd = params["guest_cmd"]
+        deps_dir = virttest_data_dir.get_deps_dir()
+        host_file = os.path.join(deps_dir, host_script)
+        try:
+            vm.copy_files_to(host_file, guest_dir)
+            session.cmd_output(params["guest_tool_install"], timeout=240)
+            session.cmd_output("chmod 755 %s" % guest_cmd)
+        except Exception as e:
+            test.fail("Guest test preperation fail: %s" % str(e))
+        status, output = session.cmd_status_output(guest_cmd, timeout=360)
+        if status:
+            test.fail("Guest script error")
```

Comment:
use cmd_status if you don't want an output, or print out the output in the fail()

Reply:
Thanks Zhenchao, updated. 
 (1/3) Host_RHEL.m9.u6.v0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.snp_basic_config.policy_default.q35: FAIL: Guest script error (167.72 s)